tidb-lightning/tidb-lightning-log-message-explanation.md

```diff
@@ -0,0 +1,360 @@
+```
+
+[table_restore.go:927](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/lightning/restore/table_restore.go#L927)，[backend.go:452](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/lightning/backend/backend.go#L452)：开始将存储在引擎中的 KV 对[导入](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/lightning/backend/local/local.go#L1311)到目标 TiKV 节点。
+
+```
+[INFO] [local.go:1023] ["split engine key ranges"] [engine=d173bb2e-b753-5da9-b72e-13a49a46f5d7] [totalSize=2159933993] [totalCount=10000000] [firstKey=74800000000000003F5F728000000000000001] [lastKey=74800000000000003F5F728000000000989680] [ranges=22]
+```
+
+[local.go:1023](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/lightning/backend/local/local.go#L1023)：在[导入引擎](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/lightning/backend/local/local.go#L1331)前，根据 [`RegionSplitSize`](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/lightning/restore/table_restore.go#L927) 配置项将引擎数据逻辑拆分为更小的范围。
+
+```
+[INFO] [local.go:1336] ["start import engine"] [uuid=d173bb2e-b753-5da9-b72e-13a49a46f5d7] [ranges=22] [count=10000000] [size=2159933993]
+```
+
+[local.go:1336](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/lightning/backend/local/local.go#L1336)：开始通过拆分范围将 KV 对导入引擎中。
+
+```
+[INFO] [localhelper.go:89] ["split and scatter region"] [minKey=7480000000000000FF3F5F728000000000FF0000010000000000FA] [maxKey=7480000000000000FF3F5F728000000000FF9896810000000000FA] [retry=0]
+```
+
+[localhelper.go:89](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/lightning/backend/local/localhelper.go#L89)：开始根据引擎范围的最小键和最大键[拆分和打散](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/lightning/backend/local/localhelper.go#L65) TiKV region。
+
+```
+[INFO] [localhelper.go:108] ["paginate scan regions"] [count=1] [start=7480000000000000FF3F5F728000000000FF0000010000000000FA] [end=7480000000000000FF3F5F728000000000FF9896810000000000FA]
+[INFO] [localhelper.go:116] ["paginate scan region finished"] [minKey=7480000000000000FF3F5F728000000000FF0000010000000000FA] [maxKey=7480000000000000FF3F5F728000000000FF9896810000000000FA] [regions=1]
+```
+
+[localhelper.go:108](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/lightning/backend/local/localhelper.go#L108)，[localhelper.go:116](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/lightning/backend/local/localhelper.go#L108)：在 PD 上分页[扫描一批 region 信息](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/restore/split.go#L413)。
+
+```
+[INFO] [split_client.go:460] ["checking whether need to scatter"] [store=1] [max-replica=3]
+[INFO] [split_client.go:113] ["skipping scatter because the replica number isn't less than store count."]
+```
+
+[split_client.go:460](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/restore/split_client.go#L460), [split_client.go:113](https://github.com/pingcap/tidb/blob/v5.4.0/br/pkg/restore/split_client.go#L113)：因为 max-replica <= TiKV store 数量，所以跳过打散 region 阶段。打散 region 是指 PD 调度器将 region 和副本分散到不同的 TiKV store 的过程。
```

Comment:
上面的 log 里写的是 “skipping scatter because the replica number isn't less than store count”，这和下面的 <= 是否矛盾
 “因为 max-replica <= TiKV store 数量，所以跳过打散 region 阶段。”

Reply:
@lance6716 PTAL

Reply:
确实写反了，应该是 max-replica >= TiKV store 数量